account_invoice_search_by_reference/models/account_move.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2019 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    @api.model
+    def name_search(self, name, args=None, operator="ilike", limit=100):
+        args = args or []
+        domain = []
+        if name:
+            domain = ["|", ("ref", operator, name), ("name", operator, name)]
+        invoices = self.search(domain + args, limit=limit)
+        return invoices.name_get()
```

Comment:
this should already be covered by core `_rec_names_search = ['name', 'partner_id.name', 'ref']`